src/textual/widgets/_markdown.py

```diff
@@ -500,19 +501,35 @@ class MarkdownFence(MarkdownBlock):
     def __init__(self, markdown: Markdown, code: str, lexer: str) -> None:
         self.code = code
         self.lexer = lexer
+        self.theme = "solarized-dark" if self.app.dark else "solarized-light"
         super().__init__(markdown)
 
-    def compose(self) -> ComposeResult:
-        from rich.syntax import Syntax
+    def _retheme(self) -> None:
+        """Swap between a dark and light theme when the mode changes."""
+        self.theme = "solarized-dark" if self.app.dark else "solarized-light"
+        code_block = self.query_one(Static)
+        code_block.renderable = Syntax(
+            self.code,
+            lexer=self.lexer,
+            word_wrap=False,
+            indent_guides=True,
+            padding=(1, 2),
+            theme=self.theme,
+        )
 
+    def _on_mount(self, _: Mount) -> None:
+        """Watch app theme switching."""
+        self.watch(self.app, "dark", self._retheme)
+
+    def compose(self) -> ComposeResult:
         yield Static(
             Syntax(
                 self.code,
                 lexer=self.lexer,
                 word_wrap=False,
                 indent_guides=True,
                 padding=(1, 2),
-                theme="material",
+                theme=self.theme,
```

Comment:
I note that the theme that was used was `"material"`. but this PR changes the theme used for dark mode; was there a reason for this?

Reply:
No real reason. I just thought that there is some "consistency" in choosing a theme that has both light and dark versions. But I would be happy to align with some other choice if there is one. What would you choose?

Reply:
For now at least I think I'd stick with "material" as the default for dark mode, so then we're not changing the look of people's applications when they're in dark mode.

Reply:
Would you have a preference for the light theme default?

Reply:
I ended up using "material-light" for consistency.

Reply:
Sounds like a good choice to me!